### PR TITLE
Change all instances of == to === and != to !==

### PR DESCRIPTION
**Pull request by tohhsinpei**
### Description
Does what it says on the tin. Exception is made for `== null` and `!= null`; see go/tsstyle#equality-checks.

We also add a new [linter rule](https://eslint.org/docs/rules/eqeqeq) enforcing this.

### Scenarios Tested
Added new instance of `foo == bar` and verified that `npm run lint:ts` gives an error. Also verified that linter does not complain about existing instances of `== null` and `!= null`.

### CODE REVIEW

File: scripts/emulator-tests/fixtures.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ rules_version = '2';
 service firebase.storage {
   match /b/{bucket}/o {
     match /{allPaths=**} {
-      allow read, write: if math.abs(-10) == 10;
+      allow read, write: if math.abs(-10) === 10;
```
Comment (tonyjhuang) [inline scripts/emulator-tests/fixtures.ts:140]: I'm not sure if `===` is valid in security rules, does this pass integration tests?

File: src/emulator/auth/schema.ts [hunk 2]
```diff
@@ -2867,7 +2867,7 @@ export interface components {
...
+     * Represents a textual expression in the Common Expression Language (CEL) syntax. CEL is a C-like expression language. The syntax and semantics of CEL are documented at https://github.com/google/cel-spec. Example (Comparison): title: "Summary size limit" description: "Determines if a summary is less than 100 chars" expression: "document.summary.size() < 100" Example (Equality): title: "Requestor is owner" description: "Determines if requestor is the document owner" expression: "document.owner === request.auth.claims.email" Example (Logic): title: "Public documents" description: "Determine whether the document should be publicly visible" expression: "document.type !== 'private' && document.type !== 'internal'" Example (Data Manipulation): title: "Notification string" description: "Create a notification string with a timestamp." expression: "'New message received at ' + string(document.create_time)" The exact variables and functions that may be referenced within an expression are determined by the service that evaluates it. See the service documentation for additional information.
```
Comment (tonyjhuang) [inline src/emulator/auth/schema.ts:2870]: you probably want to revert this

File: src/init/features/hosting/github.ts [hunk 3]
```diff
@@ -298,7 +298,7 @@ function writeChannelActionYMLFile(
     on: "pull_request",
     jobs: {
       ["build_and_preview"]: {
-        if: "${{ github.event.pull_request.head.repo.full_name == github.repository }}", // secrets aren't accessible on PRs from forks
+        if: "${{ github.event.pull_request.head.repo.full_name === github.repository }}", // secrets aren't accessible on PRs from forks
```
Comment (tonyjhuang) [inline src/init/features/hosting/github.ts:301]: revert

File: src/test/emulators/fixtures.ts [hunk 4]
```diff
@@ -166,7 +166,7 @@ service firebase.storage {
       allow read: if true
     }
     match /testing/{allPaths=**} {
-      allow write: if request.auth.token.firebase.sign_in_provider == 'anonymous';
+      allow write: if request.auth.token.firebase.sign_in_provider === 'anonymous';
```
Comment (tonyjhuang) [inline src/test/emulators/fixtures.ts:169]: revert file

File: src/test/emulators/storage.rules.spec.ts [hunk 5]
```diff
@@ -152,7 +152,7 @@ describe.skip("Storage Rules", function () {
             rules_version = '2';
             service firebase.storage {
               match /b/{bucket}/o/{sizeSegment=**} {
-                allow read: if request.auth.uid == 'mock-user';
+                allow read: if request.auth.uid === 'mock-user';
```
Comment (tonyjhuang) [inline src/test/emulators/storage.rules.spec.ts:155]: revert

**Review by tonyjhuang — COMMENTED** [review]
Mostly LGTM but I imagine you'll want to hunt down approvals for the files we don't own. For example, we should have someone from the Firebase Hosting team stamp approval for their files separately

File: scripts/emulator-tests/fixtures.ts (hunk 1 above)
Comment (tohhsinpei) [reply scripts/emulator-tests/fixtures.ts:140]: Done.

File: src/test/emulators/storage.rules.spec.ts (hunk 5 above)
Comment (tohhsinpei) [reply src/test/emulators/storage.rules.spec.ts:155]: Done.

File: src/test/emulators/fixtures.ts (hunk 4 above)
Comment (tohhsinpei) [reply src/test/emulators/fixtures.ts:169]: Done.

File: src/init/features/hosting/github.ts (hunk 3 above)
Comment (tohhsinpei) [reply src/init/features/hosting/github.ts:301]: Done.

File: src/emulator/auth/schema.ts (hunk 2 above)
Comment (tohhsinpei) [reply src/emulator/auth/schema.ts:2870]: Done.

**Review by tonyjhuang — APPROVED** [review]
LGTM for storage files, please wait for approvals from other file owners before submitting

File: .eslintrc.js [hunk 6]
```diff
@@ -63,6 +63,7 @@ module.exports = {
         "no-fallthrough": "warn", // TODO(bkendall): remove, allow to error.
         "no-unused-vars": "warn", // TODO(bkendall): remove, allow to error.
         camelcase: ["warn", { ignoreDestructuring: true }], // TODO(bkendall): remove, allow to error.
+        eqeqeq: ["error", "always", { null: "ignore" }],
```
Comment (bkendall) [inline .eslintrc.js:66]: If this is only for `ts` files, this config should go in the block with `no-invalid-this`, grouping it with the other "permanent" rules we use. If this should be for _all_ files, put it in the first block under `rules`

**Review by bkendall — APPROVED** [review]
Looks good to me. One adjustment in .eslintrc.js, but should be easy enough.

File: .eslintrc.js (hunk 6 above)
Comment (tohhsinpei) [reply .eslintrc.js]: Done, moved to block with permanent rules.

**Pull request merged**